runtime-modules/content/src/lib.rs

```diff
@@ -2416,6 +2483,65 @@ impl<T: Trait> Module<T> {
             video_id, member_id,
         ))
     }
+
+    fn ensure_can_claim_channel_reward(
+        origin: &T::Origin,
+        actor: &ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
+        item: &PullPayment<T>,
+        proof: &[ProofElement<T>],
+    ) -> Result<ChannelRewardClaimInfo<T>, DispatchError> {
+        let channel = Self::ensure_channel_exists(&item.channel_id)?;
+
+        ensure_actor_authorized_to_claim_payment::<T>(origin.clone(), &actor, &channel.owner)?;
+
+        ensure!(
+            Self::channel_cashouts_enabled(),
+            Error::<T>::ChannelCashoutsDisabled
+        );
+
+        let reward_account = Self::ensure_reward_account(&channel)?;
+
+        let cashout = item
+            .cumulative_reward_earned
+            .saturating_sub(channel.cumulative_reward_claimed);
+
+        ensure!(
+            <MaxCashoutAllowed<T>>::get() >= cashout,
+            Error::<T>::CashoutAmountExceedsMaximumAmount
+        );
+        ensure!(
+            <MinCashoutAllowed<T>>::get() <= cashout,
+            Error::<T>::CashoutAmountBelowMinimumAmount
+        );
+        Self::verify_proof(proof, item)?;
+        Ok((channel, reward_account, cashout))
+    }
+
+    fn execute_channel_reward_claim(
+        channel_id: T::ChannelId,
+        reward_account: &T::AccountId,
+        amount: BalanceOf<T>,
+    ) {
+        ContentTreasury::<T>::transfer_reward(reward_account, amount);
+        ChannelById::<T>::mutate(&channel_id, |channel| {
+            channel.cumulative_reward_claimed =
+                channel.cumulative_reward_claimed.saturating_add(amount)
+        });
+    }
+
+    fn execute_channel_balance_withdrawal(
+        reward_account: &T::AccountId,
+        destination: &T::AccountId,
+        amount: BalanceOf<T>,
+    ) -> DispatchResult {
+        <Balances<T> as Currency<T::AccountId>>::transfer(
+            reward_account,
+            destination,
+            amount,
+            ExistenceRequirement::AllowDeath,
```

Comment:
Should be changed to KeepAlive: https://github.com/Joystream/joystream/issues/3497

Reply:
https://github.com/Joystream/joystream/issues/3497#issuecomment-1080531129

Reply:
Ok